File: src/services/eventStore.js
class EventStore {
  constructor(maxItems) {
    this.maxItems = maxItems;
    this.items = [];
  }

  add(event) {
    this.items.unshift(event);
    if (this.items.length > this.maxItems) {
      this.items.pop();
    }
  }

  list() {
    return this.items;
  }

  count() {
    return this.items.length;
  }
}

module.exports = EventStore;

File: fast-api/app/services/event_store.py
from collections import deque
from typing import Deque, List

from app.models.ppe import StoredEvent


class EventStore:
    def __init__(self, max_items: int) -> None:
        self._items: Deque[StoredEvent] = deque(maxlen=max_items)

    def add(self, event: StoredEvent) -> None:
        self._items.appendleft(event)

    def list(self) -> List[StoredEvent]:
        return list(self._items)

    def count(self) -> int:
        return len(self._items)
